server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -651,7 +657,10 @@ private Repository createRepository(RepositoryMetadata repositoryMetadata, Map<S
         logger.debug("creating repository [{}][{}]", repositoryMetadata.type(), repositoryMetadata.name());
         Repository.Factory factory = factories.get(repositoryMetadata.type());
         if (factory == null) {
-            throw new RepositoryException(repositoryMetadata.name(), "repository type [" + repositoryMetadata.type() + "] does not exist");
+            throw new RepositoryPluginException(
+                repositoryMetadata.name(),
+                "repository type [" + repositoryMetadata.type() + "] does not exist"
+            );
```

Comment:
```
[2022-01-12T09:22:56,827][WARN ][o.e.r.RepositoriesService] [idegtiarenko-ThinkPad-T490] failed to create repository [repository-hdfs]
org.elasticsearch.repositories.RepositoryPluginException: [repository-hdfs] repository type [hdfs] does not exist
	at org.elasticsearch.repositories.RepositoriesService.createRepository(RepositoriesService.java:662) ~[elasticsearch-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
	at org.elasticsearch.repositories.RepositoriesService.applyClusterState(RepositoriesService.java:505) [elasticsearch-8.1.0-SNAPSHOT.jar:8.1.0-SNAPSHOT]
```

Should I reword this to `repository plugin for type [type] does not exist` or keep existing message?

Reply:
Blaming a missing plugin here seems wrong, it could just be a typo, but if `createRepository` knew whether to fail like this or not then I'd suggest the same message as I suggested above:

`"repository type [" + repositoryMetadata.type() + "] is unknown; ensure that all required plugins are installed on this node"`